sdk/eventhubs/azure_messaging_eventhubs/examples/eventhubs_tracing_subscriber.rs

```diff
@@ -0,0 +1,217 @@
+};
+
+/// A custom tracing layer that sends log messages to Azure Event Hubs.
+struct EventHubsLayer {
+    sender: mpsc::Sender<EventData>,
+}
+
+impl EventHubsLayer {
+    async fn new(
+        fully_qualified_namespace: &str,
+        eventhub_name: &str,
+    ) -> Result<Self, Box<dyn Error>> {
+        let producer = ProducerClient::builder()
+            .open(
+                fully_qualified_namespace,
+                eventhub_name,
+                DefaultAzureCredential::new()?,
+            )
+            .await?;
+
+        let (sender, mut receiver) = mpsc::channel::<EventData>(100);
+        let producer_arc = Arc::new(producer);
+        let producer_clone = producer_arc.clone();
+
+        // Spawn a task that receives log messages and sends them to EventHubs
+        tokio::spawn(async move {
+            while let Some(event) = receiver.recv().await {
+                println!("Sending event to EventHubs: {:?}", event);
+                if let Err(err) = producer_clone.send_event(event, None).await {
+                    eprintln!("Failed to send event to EventHubs: {err}");
+                }
+            }
+        });
+
+        Ok(Self { sender })
+    }
+}
+
+/// Constants for property names in the EventData.
+/// These constants are used to set properties in the EventData object.
+const LOG_LEVEL_PROPERTY: &str = "log_level";
+const NAME_PROPERTY: &str = "name";
+const TARGET_PROPERTY: &str = "target";
+const MODULE_PATH_PROPERTY: &str = "module_path";
+const FILE_PROPERTY: &str = "file";
+const LINE_PROPERTY: &str = "line";
+const FIELDS_PROPERTY: &str = "fields";
+const IS_SPAN_PROPERTY: &str = "is_span";
+const IS_EVENT_PROPERTY: &str = "is_event";
+const TIMESTAMP_PROPERTY: &str = "timestamp";
+
+impl<S> Layer<S> for EventHubsLayer
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+{
+    fn on_event(&self, event: &Event<'_>, _ctx: Context<'_, S>) {
+        // Format the event into a string
+        let mut visitor = EventVisitor {
+            buffer: String::new(),
+        };
+
+        event.record(&mut visitor);
+
+        // Add timestamp, level, and target
+        let metadata = event.metadata();
+
+        // Build an EventData object containing both the event body and properties reflecting the event's metadata.
+        let mut event_data_builder = EventData::builder()
+            .with_body(visitor.buffer)
+            .add_property(LOG_LEVEL_PROPERTY.to_string(), metadata.level().as_str())
+            .add_property(
+                TIMESTAMP_PROPERTY.to_string(),
+                SystemTime::now()
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .unwrap()
+                    .as_secs(),
+            )
+            .add_property(TARGET_PROPERTY.to_string(), metadata.target())
+            .add_property(NAME_PROPERTY.to_string(), metadata.name())
+            .add_property(FIELDS_PROPERTY.to_string(), metadata.fields().to_string())
```

Comment:
There's no hierarchical support for fields?

Reply:
Not currently, and I think I should add it.

Reply:
Adding them to the property bag.